azurerm/internal/services/iothub/iothub_resource.go

```diff
@@ -346,6 +346,38 @@ func resourceArmIotHub() *schema.Resource {
 				},
 			},
 
+			"enrichment": {
+				Type: schema.TypeList,
+				// Currently only 10 enrichments is allowed for standard or basic tier, 2 for Free tier.
+				MaxItems:   10,
+				Optional:   true,
+				Computed:   true,
+				ConfigMode: schema.SchemaConfigModeAttr,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringMatch(
+								regexp.MustCompile("^[-_.a-zA-Z0-9]{1,64}$"),
+								"Enrichment Key name can only include alphanumeric characters, periods, underscores, hyphens, has a maximum length of 64 characters, and must be unique.",
+							),
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"endpoint_names": {
+							Type: schema.TypeList,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
```

Comment:
And validate these?

Reply:
I do the same as it is done for [route endpoints](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/iothub/iothub_resource.go#L334)